webrender_traits/src/webgl.rs

```diff
@@ -973,6 +994,16 @@ impl WebGLCommand {
         chan.send(shader).unwrap();
     }
 
+    fn create_vertex_array(gl: &gl::Gl, chan: MsgSender<Option<WebGLVertexArrayId>>) {
+        let vao = gl.gen_vertex_arrays(1)[0];
+        let vao = if vao == 0 {
```

Comment:
> Shouldn't it be forbidden to do any GL calls in this case?

Yes. It must be checked in the Servo WebGL implementation. It's possible to call WebGL methods when a context is lost (e.g. a notification when the app is in background) but those calls should be discarted by the WebGLRendeirngContext and don't arrive here.

> That would be a GL error then. What is our WebGL implementation policy with regards to GL errors?

Most of the errors are detected without calling the driver following the rules defines in WebGL spec. Other than that,  the WebGL developer should be responsible for calling getError() and act accordingly.

> Thanks for pointing this out. I'd like to investigate further.

The WebGL spec uses nullable return values for createTexture, createFramebuffer, etc. One of the reason is because of the possibility of calling the methods with a lost context. I don't find anything about other kind of errors to return null values. I think that @emilio implemented the check for create_xxx calls, do you know any other case where it can be 0/null is a valid value?